### PR TITLE
Add utils for ec2_placement_group

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### SUMMARY

This PR adds utils for ec2_placement_group

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: changelogs/fragments/add_utils_ec2_placement_group.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - module_utils/ec2 - add utils for the ec2_placement_group* modules.
```
Comment (abikouo) [inline changelogs/fragments/add_utils_ec2_placement_group.yml:2]:
```suggestion
  - module_utils/ec2 - add utils for the ec2_placement_group* modules (https://github.com/ansible-collections/amazon.aws/pull/2322).
```

File: plugins/module_utils/ec2.py [hunk 2]
```diff
@@ -1241,6 +1241,39 @@ def replace_network_acl_association(client, network_acl_id: str, association_id:
...
+    return client.describe_placement_groups(**params)["PlacementGroups"]
+
+
+@EC2PlacementGroupErrorHandler.deletion_error_handler("delete placement group")
+@AWSRetry.jittered_backoff()
+def delete_placement_group(client, group_name: Optional[str] = None) -> bool:
+    params = {}
+    if group_name:
+        params["GroupName"] = group_name
+    client.delete_placement_group(**params)
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1267]:
```suggestion
def delete_placement_group(client, group_name: str) -> bool:
    client.delete_placement_group(GroupName=group_name)
```
The `GroupName` parameter is required to delete the placement group

File: plugins/module_utils/ec2.py [hunk 3]
```diff
@@ -1241,6 +1241,39 @@ def replace_network_acl_association(client, network_acl_id: str, association_id:
...
+    return True
+
+
+@EC2PlacementGroupErrorHandler.common_error_handler("create placement group")
+@AWSRetry.jittered_backoff()
+def create_placement_group(client, **params: Dict[str, Union[str, EC2TagSpecifications]]) -> Dict[str, Any]:
+    return client.create_placement_group(**params)["PlacementGroups"]
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1274]:
```suggestion
    return client.create_placement_group(**params)["PlacementGroup"]
```

**Review by abikouo — CHANGES_REQUESTED** [review]

**Review by alinabuzachis — APPROVED** [review]
I'm not sure if it makes sense to use DryRun. I would probably use check_mode without passing DryRun, but I would also like to know what @tremble thinks about.

File: plugins/module_utils/ec2.py [hunk 4]
```diff
@@ -1241,6 +1241,40 @@ def replace_network_acl_association(client, network_acl_id: str, association_id:
...
+def describe_ec2_placement_groups(
+    client, **params: Dict[str, Union[List[str], int, List[Dict[str, Union[str, List[str]]]]]]
+) -> List[Dict[str, Any]]:
+    return client.describe_placement_groups(**params)["PlacementGroups"]
+
+
+@EC2PlacementGroupErrorHandler.deletion_error_handler("delete placement group")
+@AWSRetry.jittered_backoff()
+def delete_ec2_placement_group(client, group_name: str, dry_run: Optional[bool] = False) -> bool:
+    client.delete_placement_group(GroupName=group_name, DryRun=dry_run)
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:1268]:
```suggestion
def delete_ec2_placement_group(client, group_name: str) -> bool:
    client.delete_placement_group(GroupName=group_name)
```
we are not using `dry_run` to run the check mode

File: plugins/module_utils/errors.py [hunk 5]
```diff
@@ -40,6 +40,8 @@ def wrapper(func):
             def handler(*args, **kwargs):
                 try:
                     return func(*args, **kwargs)
+                except cls._is_dry_run():
+                    return {"boto3_error_code": "DryRunOperation"}
```
Comment (abikouo) [inline plugins/module_utils/errors.py:44]:
```suggestion
```

**Review by abikouo — CHANGES_REQUESTED** [review]

**Comment by GomathiselviS** [comment]
> I'm not sure if it makes sense to use DryRun. I would probably use check_mode without passing DryRun, but I would also like to know what @tremble thinks about.

I've currently removed it. From my assessment based on the definition of check_mode, the API should not be invoked at all. Moreover, DryRun is utilized across various AWS APIs beyond just placement groups. Given that other modules haven't implemented it, it seemed logical to exclude the DryRun functionality from this particular module.

**Pull request merged**